pkg/controller/acmeorders/sync.go

```diff
@@ -144,10 +152,25 @@ func (c *controller) Sync(ctx context.Context, o *cmacme.Order) (err error) {
 		return err
 	}
 
+	acmeOrder, err := getACMEOrder(ctx, cl, o)
```

Comment:
Just stumbled across this - so one thing to note is that this controller (and all other acme controllers) have always made a point to avoid calling the ACME API unless it is _strictly_ necessary. This is because, in instances where unexpected errors can crop up (or a bug in our code), it can lead to large amounts of requests being sent to Let's Encrypt from many different cert-manager installations. That is also why you'll notice very few calls using the ACME client throughout these controllers.

If there is no other way to achieve this without calling out to the ACME API, then that's fine - but this needs a lot of care to ensure there's no cases where we can enter into loops. It's also worth viewing it as a last resort and an expensive call to make, so gating this kind of call behind as many local checks as possible would make sense (eg given the 'happy path' today works *without* making this extra call, is there an error case that we could add this call into rather than having it on the happy path).

By adding it on the happy path of all syncs of orders, we have increased the number of api calls made in this controller for _everyone_ by 1